### PR TITLE
Fix: renamed DefaultKubernetesDomain to DefaultClusterLocalDomain

### PR DESCRIPTION
**Pull request by ramanujadasu**
**Please provide a description of this PR:**
Renamed DefaultKubernetesDomain to DefaultClusterLocalDomain as mentioned TODO in the **istio/pkg/config/constants/constants.go**

### CODE REVIEW

File: releasenotes/notesRelease/38597.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: feature
+area: traffic-management
+issue:
+  - 38597
+releaseNotes:
+- |
+  **Fixed** Renamed DefaultKubernetesDomain to DefaultClusterLocalDomain as mentioned TODO in the istio/pkg/config/constants/constants.go
```
Comment (nmittler) [inline releasenotes/notesRelease/38597.yaml:8]: Don't need a release note for this ... just add a `release-notes-none` label to the PR to get around the check.

File: pkg/config/constants/constants.go [hunk 2]
```diff
@@ -59,10 +59,9 @@ const (
 	// IstioIngressNamespace is the namespace where Istio ingress controller is deployed
 	IstioIngressNamespace = "istio-system"
 
-	// DefaultKubernetesDomain the default service domain suffix for Kubernetes, if not overridden in config.
-	// TODO(nmittler): Rename this to DefaultClusterLocalDomain.
-	// TODO(nmittler): Search/replace explicit usages of the string with this constant.
-	DefaultKubernetesDomain = "cluster.local"
+	// DefaultClusterLocalDomain the default service domain suffix for Kubernetes, if not overridden in config.
+	// TODO(nmittler): Renamed DefaultKubernetesDomain to DefaultClusterLocalDomain.
```
Comment (nmittler) [inline pkg/config/constants/constants.go:63]: remove the TODO

**Review by nmittler — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Pull request merged**